backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -710,52 +683,46 @@ public UserInfo getUserByLoginEmail(String loginEmail) {
         throw new RestrictedAccessUserException();
       }
 
-      return consolidateUser(
-          apiUser,
-          oktaUser.getOrganizationRoleClaims(),
-          oktaUser.getStatus(),
-          oktaUser.isSiteAdmin());
+      return consolidateUser(apiUser, oktaUser);
     } catch (IllegalGraphqlArgumentException | UnidentifiedUserException e) {
       throw new OktaAccountUserException();
     }
   }
 
-  private UserInfo consolidateUser(
-      ApiUser apiUser,
-      Optional<OrganizationRoleClaims> optClaims,
-      UserStatus userStatus,
-      boolean isSiteAdmin) {
-
-    OrganizationRoleClaims claims = optClaims.orElseThrow(UnidentifiedUserException::new);
-
-    // use the target user's org so response is built correctly even if site admin is the requester
-    Organization org = _orgService.getOrganization(claims.getOrganizationExternalId());
+  private OrganizationRoles getOrganizationRoles(
+      Optional<OrganizationRoleClaims> oktaClaims, ApiUser apiUser, boolean isSiteAdmin) {
+    OrganizationRoles orgRoles = null;
+    if (oktaClaims.isPresent()) {
+      orgRoles = _orgService.getOrganizationRoles(oktaClaims.get());
+    }
 
-    List<Facility> facilities = _orgService.getFacilities(org);
-    Set<Facility> facilitiesSet = new HashSet<>(facilities);
+    if (!isSiteAdmin) {
+      if (_featureFlagsConfig.isOktaMigrationEnabled()) {
+        orgRoles = getOrgRolesFromDB(apiUser);
+      } else {
+        setRolesAndFacilities(Optional.ofNullable(orgRoles), apiUser);
+      }
+    }
 
-    boolean allFacilityAccess = claims.grantsAllFacilityAccess();
-    Set<Facility> accessibleFacilities =
-        allFacilityAccess
-            ? facilitiesSet
-            : facilities.stream()
-                .filter(f -> claims.getFacilities().contains(f.getInternalId()))
-                .collect(Collectors.toSet());
+    return orgRoles;
+  }
 
+  private UserInfo consolidateUser(ApiUser apiUser, PartialOktaUser oktaUser) {
+    boolean isSiteAdmin = oktaUser.isSiteAdmin();
+    UserStatus userStatus = oktaUser.getStatus();
+    OrganizationRoleClaims oktaClaims =
+        oktaUser.getOrganizationRoleClaims().orElseThrow(UnidentifiedUserException::new);
     OrganizationRoles orgRoles =
-        new OrganizationRoles(org, accessibleFacilities, claims.getGrantedRoles());
-    if (!_featureFlagsConfig.isOktaMigrationEnabled() && !isSiteAdmin) {
-      try {
-        setRolesAndFacilities(orgRoles, apiUser);
-      } catch (PrivilegeUpdateFacilityAccessException e) {
-        log.warn(
-            "Could not migrate roles and facilities for user with id={} because facilities were invalid",
-            apiUser.getInternalId());
-      }
-    }
+        getOrganizationRoles(Optional.ofNullable(oktaClaims), apiUser, isSiteAdmin);
+
     return new UserInfo(apiUser, Optional.of(orgRoles), isSiteAdmin, userStatus);
```

Comment:
doesn't this return a `UserInfo` that contains the Okta role/facility info? when we do this are we subbing in our DB info somewhere else in the code?

Reply:
the `getOrganizationRoles` method that was [added here](https://github.com/CDCgov/prime-simplereport/pull/8030/files/cbf0a576442c1d93127eb6ec95a324a1041ce92f#diff-3a4cd84e094d22aa2350c1d2c5c0a1a66e1a4952f73b13ef86654c7e2d285477R692) handles that logic now. Let me know if this can be more clear by either renaming the method or something else! 👀

Reply:
no that makes sense! it came together in my head now that I read more, thank you! sorry for missing that

Reply:
No worries at all! Thank you for asking :D